src/cobra/core/reaction.py

```diff
@@ -411,46 +411,23 @@ def metabolites(self):
     def genes(self):
         return frozenset(self._genes)
 
-    @property
-    def gene_reaction_rule(self):
-        return self._gene_reaction_rule
-
-    @gene_reaction_rule.setter
-    def gene_reaction_rule(self, new_rule):
-
-        # TODO: Do this :)
-        if get_context(self):
-            warn("Context management not implemented for " "gene reaction rules")
-
-        self._gene_reaction_rule = new_rule.strip()
-        try:
-            _, gene_names = parse_gpr(self._gene_reaction_rule)
-        except (SyntaxError, TypeError) as e:
-            if "AND" in new_rule or "OR" in new_rule:
-                warn(
-                    "uppercase AND/OR found in rule '%s' for '%s'"
-                    % (new_rule, repr(self))
-                )
-                new_rule = uppercase_AND.sub("and", new_rule)
-                new_rule = uppercase_OR.sub("or", new_rule)
-                self.gene_reaction_rule = new_rule
-                return
-            warn("malformed gene_reaction_rule '%s' for %s" % (new_rule, repr(self)))
-            tmp_str = and_or_search.sub("", self._gene_reaction_rule)
-            gene_names = set((gpr_clean.sub(" ", tmp_str).split(" ")))
-        if "" in gene_names:
-            gene_names.remove("")
-        old_genes = self._genes
+    def update_genes_from_gpr(self, new_gene_names: set = None):
```

Comment:
make this a private or hidden method (start the name with `__` or `_`).

Reply:
Done